pkg/util/utils.go

```diff
@@ -316,6 +316,7 @@ func GetDefaultStoreOptions() (storage.StoreOptions, string, error) {
 
 		storageConf := filepath.Join(os.Getenv("HOME"), ".config/containers/storage.conf")
 		if _, err := os.Stat(storageConf); err == nil {
+			storageOpts = storage.StoreOptions{}
```

Comment:
I have a hard time wrapping my head around why this function behaves so very different in rootless-mode. To me, it seems that the code for storage configuration should be centralised and refactored. I think that this code belongs into containers/storage with an API that Buildah and others can use as well.

Reply:
the main reason was to not have any logic related to fuse-overlayfs in containers/storage.  I personally think it was a mistake to abstract fuse-overlayfs into `mount_program` for the overlay driver instead of just dealing with one implementation (I seriously doubt anyone is interested in reimplementing fuse-overlayfs and make it compatible with containers/storage).